frontend/encore/dev-server.rst

```diff
@@ -66,53 +51,65 @@ method in your ``webpack.config.js`` file:
 
     The ``Encore.configureDevServerOptions()`` method was introduced in Encore 0.28.4.
 
-Hot Module Replacement HMR
---------------------------
+Enabling HTTPS using the Symfony Web Server
+-------------------------------------------
+
+If you're using the :doc:`Symfony web server </setup/symfony_server>` locally with HTTPS,
+you'll need to also tell the dev-server to use HTTPS. To do this, you can reuse the Symfony web
+server SSL certificate:
+
+.. code-block:: javascript
 
-Encore *does* support `HMR`_ for :doc:`Vue.js </frontend/encore/vuejs>`, but
-does *not* work for styles anywhere at this time. To activate it, pass the ``--hot``
-option:
+    // webpack.config.js
+    // ...
+    const path = require('path');
+
+    Encore
+        // ...
+
+        .configureDevServerOptions(options => {
+            options.https = {
+                pfx: path.join(process.env.HOME, '.symfony/certs/default.p12'),
+            }
+        })
+
+Then make sure you run the ``dev-server`` with the ``--https`` option:
 
 .. code-block:: terminal
 
-    $ ./node_modules/.bin/encore dev-server --hot
+    $ yarn encore dev-server --https
 
-If you want to use SSL with self-signed certificates, add the ``--https``,
-``--pfx=``, and  ``--allowed-hosts`` options to the ``dev-server`` command in
-the ``package.json`` file:
+CORS Issues
+-----------
 
-.. code-block:: diff
+If you experience issues related to CORS (Cross Origin Resource Sharing), set
+the ``firewall`` option:
 
-    {
-        ...
-        "scripts": {
-    -        "dev-server": "encore dev-server",
-    +        "dev-server": "encore dev-server --https --pfx=$HOME/.symfony/certs/default.p12 --allowed-hosts=mydomain.wip",
-            ...
-        }
-    }
+.. code-block:: javascript
 
-If you experience issues related to CORS (Cross Origin Resource Sharing), add
-the ``--disable-host-check`` and ``--port`` options to the ``dev-server``
-command in the ``package.json`` file:
+    // webpack.config.js
+    // ...
+
+    Encore
+        // ...
 
-.. code-block:: diff
+        .configureDevServerOptions(options => {
+            options.firewall = false;
+        })
 
-    {
-        ...
-        "scripts": {
-    -        "dev-server": "encore dev-server",
-    +        "dev-server": "encore dev-server --port 8080 --disable-host-check",
-            ...
-        }
-    }
+Hot Module Replacement HMR
+--------------------------
 
-.. caution::
+Hot module replacement is a superpower of the ``dev-server`` where styles and
+(in some cases) JavaScript can automatically update without needing to reload
+your page. HMR works automatically with CSS (as long as you're using the
+``dev-server`` and Encore 1.0 or higher) but only works with some JavaScript
+(like :doc:`Vue.js </frontend/encore/vuejs>`).
 
-    Beware that `it's not recommended to disable host checking`_ in general, but
-    here it's required to solve the CORS issue.
```

Comment:
I think that warning should be kept since it is still not recommended to disable those checks entirely (you can usually make it work by explicitly whitelisting hosts).

Reply:
Got it here: https://github.com/symfony/symfony-docs/pull/14929